### ISSUE TITLE
Rely on compliant licenses only

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#322]
Now we're able to [resolve .nuspec files](#316), we also can build rules involving which licenses to support. In fact with #315 (now in draft) the first steps in that direction have been made.

The most important question however is on how to configure which licenses are compliant.

From a configuration perspective, the seems ideal:

``` XML
<project>
  <ProperyGroup>
    <AllowedLicenses>MIT;Apache2.0</AllowedLicenses>
    <AllowedPackages>MyCompany.*</AllowedPackages>
  </ProperyGroup>
</project>
```

Another thing would be that I think that `MIT` and `Apache2.0` (it seems that Microsoft uses the latter for most/all of its packages) should be included if nothing is specified. Should there be others that should be included automatically?

I would argue that packages not containing a package should always be added in the list of `<AllowedPackages>`. Using packages without an explicit license is - imho - a bad practice.

But getting the right value (taking conditionals and such into account) is something we do not have fully covered. That being said, it might be worth checking: See: https://platform.uno/blog/using-msbuild-items-and-properties-in-c-9-source-generators/

**Comment by CptWesley** [comment]
In the branch I was working on I base the set of compatible licenses based on the license of the current package. This of course only works for open source licensed projects. But in principle any permissive license should most likely be compatible at all times. Viral copy-left licenses such as the (A)GPL family are only compatible with themselves. A few other similar viral licenses also exist, SPDX provides an exhaustive list of all supported package licenses. Also note that packages can be licensed under multiple license, these licenses either have to both be followed, or either can be chosen (SPDX also standardizes the expression syntax for this), which complicates the matter.

On the note of not having a license, in principle this means the package is propietary and should be avoided (since lack of open source license defaults to regular copyright laws). Why someone would just upload it to a public NuGet is a different matter.
